File: ios/Classes/ExpoQQModuleTypes.swift
//
//  ExpoQQModuleTypes.swift
//  ExpoQQ
//
//  Created by Aron on 2025/8/12.
//

import Foundation
import ExpoModulesCore

struct ShareContentOptions: Record {
    @Field
    var imageBase64OrImageUri: String = ""
    @Field
    var previewImageBase64OrImageUri: String?
    @Field
    var title: String?
    @Field
    var description: String?
    @Field
    var extraImageList: [String]?
    @Field
    var platform: String?
}


struct ShareVideoOptions: Record {
    @Field
    var url: URL?
    @Field
    var title: String?
    @Field
    var description: String?
    @Field
    var previewImageURL: URL?
    @Field
    var targetContentType: String?
    @Field
    var platform: String?
}

File: android/src/main/java/expo/modules/qq/ExpoQQModuleTypes.kt
package expo.modules.qq

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record


class ShareContentOptions: Record {
    @Field
    var imageBase64OrImageUri: String = ""
    @Field
    var previewImageBase64OrImageUri: String? = null
    @Field
    var title: String? = null
    @Field
    var description: String?  = null
    @Field
    var extraImageList: Array<String>? = null
    @Field
    var platform: String? = null

    @Field
    var appName: String? = null

    @Field
    var targetUrl: String? = null
}